pulp_ansible/app/galaxy/v3/serializers.py

```diff
@@ -200,17 +202,28 @@ def get_artifact(self, obj):
         Get atrifact summary.
         """
         content_artifact = ContentArtifact.objects.select_related("artifact").filter(content=obj)
-        return ArtifactRefSerializer(content_artifact.get()).data
+        if content_artifact.get().artifact:
+            return ArtifactRefSerializer(content_artifact.get()).data
 
     def get_download_url(self, obj) -> str:
         """
         Get artifact download URL.
         """
-        host = settings.ANSIBLE_CONTENT_HOSTNAME.strip("/")
-        distro_base_path = self.context["path"]
-        filename_path = obj.relative_path.lstrip("/")
-        download_url = f"{host}/{distro_base_path}/{filename_path}"
-        return download_url
+        content_artifact = ContentArtifact.objects.select_related("artifact").filter(content=obj)
```

Comment:
we repeat this query 3 times,
maybe we can create a cache, something similar to it: https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/app/galaxy/v3/views.py#L77-L84

Reply:
+1 to this. We've had various performance concerns in the past.